src/engine/routing_algorithms/many_to_many.cpp

```diff
@@ -72,10 +104,116 @@ void relaxOutgoingEdges(const datafacade::ContiguousInternalMemoryDataFacade<Alg
     }
 }
 
+template <bool DIRECTION>
+void relaxOutgoingEdges(
+    const datafacade::ContiguousInternalMemoryDataFacade<mld::Algorithm> &facade,
+    const NodeID node,
+    const EdgeWeight weight,
+    const EdgeDuration duration,
+    typename SearchEngineData<mld::Algorithm>::ManyToManyQueryHeap &query_heap)
+{
+    const auto &partition = facade.GetMultiLevelPartition();
+    const auto &cells = facade.GetCellStorage();
+
+    const auto &node_data = query_heap.GetData(node);
+    const auto level =
+        std::max(node_data.level, partition.GetHighestDifferentLevel(node_data.parent, node));
```

Comment:
@TheMarex thanks for pointing to the edge case! I added a similar test case to check the current implementation. In your example node `e` will not be updated in forward and will have weight 21 that corresponds a path without crossing the cell border. In the reverse pass if the starting node is `e` the meeting point will be `v`  with weight 5 and the shortest path for `s->e` will be `s->a->b->d->v->e` with weight 5+1. Using level `partition.GetHighestDifferentLevel(node, source)` instead of the current one looks good to me, I will check it on Monday.